### PR TITLE
Upgrade plugin for Drone 0.5+

### PR DESCRIPTION
**Pull request by tonglil**
Folks have tested this in Drone 0.8 for months now in the `develop` tag.

Please view the [CHANGELOG](https://github.com/NYTimes/drone-gke/blob/f9d69733f342cdf14a0bb01be253d38d7ab8cd87/CHANGELOG.md), [DOCS](https://github.com/NYTimes/drone-gke/blob/f9d69733f342cdf14a0bb01be253d38d7ab8cd87/DOCS.md), and the [README](https://github.com/NYTimes/drone-gke/blob/f9d69733f342cdf14a0bb01be253d38d7ab8cd87/README.md) for examples and instructions.

## Pre-merge

- [x] merge https://github.com/NYTimes/drone-gke/pull/65 for changelog
- [x] merge https://github.com/NYTimes/drone-gke/pull/61 for 0.4
- [x] merge https://github.com/NYTimes/drone-gke/pull/59 for 0.4
- [x] create a `0.4.x` branch for maintenance releases
- [x] announce `nytimes/drone-gke:latest` will be compatible with Drone 0.5+; for Drone 0.4, use the `nytimes/drone-gke:0.4` tags.

**This PR is slated to be merged on Feb 16, 2018.**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- Reduced available "default" environment variables to the plugin.
+To reference other environment variables, pass them via `vars`.
+  - `DRONE_BUILD_NUMBER`
+  - `DRONE_COMMIT`
+  - `DRONE_BRANCH`
+  - `DRONE_TAG`
+
+**FEATURES & IMPROVEMENTS**
+
+- `kubectl` is updated to 1.6.6.
```
Comment (yunzhu-li) [inline CHANGELOG.md]: Should we update to even newer version?

File: example/.kube.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
```
Comment (yunzhu-li) [inline example/.kube.yml]: It looks like the secrets not used in the deployment.

File: example/.kube.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        env: {{.env}}
+    spec:
+      containers:
+        - name: app
+          image: {{.image}}
+          ports:
+            - containerPort: 8080
+---
+apiVersion: v1
+kind: Service
```
Comment (yunzhu-li) [inline example/.kube.yml]: Are we updating this according to our recommendations? (i.e. Ingress)

Comment (tonglil) [reply example/.kube.yml]: No, I want a simple and runnable example across all platforms.

File: example/.kube.yml (hunk 2 above)
Comment (tonglil) [reply example/.kube.yml]: Good call.

File: CHANGELOG.md (hunk 1 above)
Comment (tonglil) [reply CHANGELOG.md]: Yes, good point.

Comment (tonglil) [reply CHANGELOG.md]: Please see https://github.com/NYTimes/drone-gke/pull/57.

File: example/.kube.yml (hunk 3 above)
Comment (yunzhu-li) [reply example/.kube.yml]: Gotcha, thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (tonglil) [reply CHANGELOG.md]: CHANGELOG updated.

**Comment by jrynyt** [comment]
hasn't this all been in production for a while now?

**Comment by bweston92** [comment]
Any idea what it is on hold for?

**Comment by tonglil** [comment]
@bweston92 we are waiting for an important change (https://github.com/NYTimes/drone-gke/pull/70) to make it in and we will tag a release for Drone 0.5+.

**Comment by tonglil** [comment]
This PR is slated to be merged on Feb 16, 2018.

File: example/.kube.yml (hunk 2 above)
Comment (tonglil) [reply example/.kube.yml]: Fixed*

**Comment by balser** [comment]
Update: This PR is slated to be merged on Feb 21, 2018.

**Comment by tonglil** [comment]
Continuous integration stopped working, I'm going to revert until we fix it.

**Pull request merged**